packages/user/Symbol/service/src/lib.rs

```diff
@@ -0,0 +1,357 @@
+            check_some(
+                Self::get(symbol_length),
+                format!("symbol length of {} not supported", symbol_length).as_str(),
+            )
+        }
+
+        pub fn add(
+            symbol_length: u8,
+            initial_price: u64,
+            target_min: u32,
+            target_max: u32,
+            floor_price: u64,
+        ) -> Self {
+            check_none(
+                Self::get(symbol_length),
+                "sale of symbol length already exists",
+            );
+
+            let nft_id = DiffAdjust::call().create(
+                initial_price,
+                86400,
+                target_min,
+                target_max,
+                floor_price,
+                50000,
+            );
+            let new_instance = Self::new(symbol_length, nft_id);
+            new_instance.save();
+            new_instance
+        }
+
+        pub fn price(&self) -> Quantity {
+            DiffAdjust::call().get_diff(self.nftId).into()
+        }
+
+        fn save(&self) {
+            SymbolLengthTable::read_write().put(&self).unwrap();
+        }
+    }
+
+    #[ComplexObject]
+    impl SymbolLength {
+        pub async fn current_price(&self) -> Decimal {
+            Decimal::new(
+                self.price(),
+                Tokens::call()
+                    .getToken(check_some(
+                        Tokens::call().getSysToken(),
+                        "system token must be defined",
+                    ))
+                    .precision,
+            )
+        }
+
+        pub async fn billing_token(&self) -> TokenRecord {
+            Tokens::call().getToken(check_some(
+                Tokens::call().getSysToken(),
+                "system token must be defined",
+            ))
+        }
+    }
+
+    #[table(name = "SymbolTable", index = 1)]
+    #[derive(Default, Fracpack, ToSchema, Serialize, Deserialize, Debug, SimpleObject)]
+    #[graphql(complex)]
+    pub struct Symbol {
+        #[primary_key]
+        pub symbolId: AccountNumber,
+        #[graphql(skip)]
+        pub ownerNft: NID,
+    }
+
+    impl Symbol {
+        fn new(symbolId: AccountNumber, ownerNft: NID) -> Self {
+            Self { symbolId, ownerNft }
+        }
+
+        pub fn get(symbol: AccountNumber) -> Option<Self> {
+            SymbolTable::read().get_index_pk().get(&symbol)
+        }
+
+        pub fn get_assert(symbol: AccountNumber) -> Self {
+            check_some(Self::get(symbol), "Symbol does not exist")
+        }
+
+        pub fn add(symbol: AccountNumber, billable: bool) -> Self {
+            check_none(Symbol::get(symbol), "Symbol already exists");
+            let length_record = SymbolLength::get(symbol.to_string().len() as u8);
+            check(
+                length_record.is_some()
+                    && symbol.to_string().chars().all(|c| c.is_ascii_lowercase()),
+                "Symbol may only contain 3 to 7 lowercase alphabetic characters",
```

Comment:
This error is inconsistent with the new implementation which isn't restricting symbol lengths from 3-7